lib/rules/keyframe-selector-notation/index.mjs

```diff
@@ -72,27 +75,18 @@ const rule = (primary) => {
 					: [];
 
 			atRuleKeyframes.walkRules((keyframeRule) => {
-				transformSelector(result, keyframeRule, (selectors) => {
-					let first = true;
+				const selectors = parseSelector(getRuleSelector(keyframeRule), result, keyframeRule);
 
-					selectors.walkTags((selectorTag) => {
-						if (first && namedTimelineRangeKeywords.has(selectorTag.value)) {
-							return false;
-						}
+				if (!selectors) return;
 
-						first = false;
+				selectors.each((selector) => {
+					const keyframeSelector = parseKeyframeSelector(selector);
 
-						checkSelector(selectorTag, optionFuncs[primary]);
-					});
-				});
+					if (!keyframeSelector) return;
+
+					const { expFunc, fixFunc } = optionFuncs[primary];
 
-				/**
-				 * @param {import('postcss-selector-parser').Tag} tag
-				 * @param {OptionFuncs} funcs
-				 */
-				function checkSelector(tag, { expFunc, fixFunc }) {
```

Comment:
`checkSelector` was only called in one place and I ran into some typing issues.

`selectors` is either `Root` or `undefined`.

From the perspective of `checkSelector` it is still possibly `undefined` even when we do a check for `undefined` in the outer scope. I don't fully understand why TypeScript considers this to be the case.

Instead of wrangling with the types or adding even more checks, I decided to inline this function as it was only called once anyway.